optimum/intel/openvino/modeling_open_clip.py

```diff
@@ -0,0 +1,680 @@
+            file_name=file_name,
+            subfolder=subfolder,
+            from_onnx=from_onnx,
+            local_files_only=local_files_only,
+            load_in_8bit=load_in_8bit,
+            quantization_config=quantization_config,
+            **kwargs,
+        )
+
+    def forward(
+        self,
+        input_ids: Union[torch.Tensor, np.ndarray],
+        **kwargs,
+    ):
+        self.compile()
+
+        inputs = {"text": input_ids}
+
+        outputs = self._inference(inputs)
+        return ModelOutput(text_features=torch.from_numpy(outputs["text_features"]))
+
+
+@add_start_docstrings(
+    """
+    OpenVINO Model for OpenCLIP Vision model for tasks like zero-shot-image-classification.
+    """,
+    MODEL_START_DOCSTRING,
+)
+class OVModelOpenCLIPVisual(OVModelOpenCLIPBase):
+    xml_model_name = "openvino_model_vision.xml"
+
+    def __init__(self, model=None, config=None, preprocess_cfg=None, **kwargs):
+        super().__init__(model, config, **kwargs)
+        self.preprocess_cfg = preprocess_cfg
+
+    @classmethod
+    def _from_transformers(
+        cls,
+        model_id: str,
+        config: PretrainedConfig,
+        use_auth_token: Optional[Union[bool, str]] = None,
+        token: Optional[Union[bool, str]] = None,
+        revision: Optional[str] = None,
+        force_download: bool = False,
+        cache_dir: str = HUGGINGFACE_HUB_CACHE,
+        subfolder: str = "",
+        local_files_only: bool = False,
+        task: Optional[str] = None,
+        trust_remote_code: bool = False,
+        load_in_8bit: Optional[bool] = None,
+        quantization_config: Union[OVWeightQuantizationConfig, Dict] = None,
+        **kwargs,
+    ):
+        if use_auth_token is not None:
+            warnings.warn(
+                "The `use_auth_token` argument is deprecated and will be removed soon. Please use the `token` argument instead.",
+                FutureWarning,
+            )
+            if token is not None:
+                raise ValueError("You cannot use both `use_auth_token` and `token` arguments at the same time.")
+            token = use_auth_token
+
+        save_dir = TemporaryDirectory()
+        save_dir_path = Path(save_dir.name)
+        # This attribute is needed to keep one reference on the temporary directory, since garbage collecting
+        # would end-up removing the directory containing the underlying OpenVINO model
+        cls._model_save_dir_tempdirectory_instance = save_dir
+
+        # If load_in_8bit and quantization_config not specified then ov_config is set to None and will be set by default in convert depending on the model size
+        if load_in_8bit is None and not quantization_config:
+            ov_config = None
+        else:
+            ov_config = OVConfig(dtype="fp32")
+
+        def fn_get_submodels(model):
+            return {"model_vision": model.visual}
+
+        custom_export_configs = {
+            "model_vision": {},
+        }
```

Comment:
question : couldn't this be inferred by `_get_open_clip_submodels_fn_and_export_configs` directly (depending on a given task) ?